src/python/tests/core/test_instance.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+import pytest
+from foglamp.core.service_registry.instance import Service
+
+__author__ = "Amarendra Kumar Sinha"
+__copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+
+class TestInstance:
+    def setup_method(self, method):
+        pass
+
+    def teardown_method(self, method):
+        Service.Instances._registry = []
+
+    @pytest.mark.asyncio
+    async def test_register(self):
+        r = Service.Instances.register("StorageService1", "Storage", "127.0.0.1", "9999")
+
+        assert "StorageService1" == r._name
+        assert "Storage" == r._type
+        assert "127.0.0.1" == r._address
+        assert "9999" == r._port
+
+    @pytest.mark.asyncio
+    async def test_register_wrong_type(self):
+        with pytest.raises(Service.InvalidServiceType) as excinfo:
+            Service.Instances.register("StorageService1", "Wrong", "127.0.0.1", "9999")
+
+        assert str(excinfo).endswith('InvalidServiceType')
+
+    @pytest.mark.asyncio
+    async def test_unregister(self):
+        r = Service.Instances.register("StorageService2", "Storage", "127.0.0.1", "8888")
+
+        assert "StorageService2" == r._name
+        assert "Storage" == r._type
+        assert "127.0.0.1" == r._address
+        assert "8888" == r._port
```

Comment:
this is already tested in ` async def test_register(self):` in next line before assigning just check r.id is a valid uuid v4